### ISSUE TITLE
[Table] column DOM doesn't always provide ID or data-column

### ISSUE THREAD

**Issue by stuckless** [IBM/carbon-addons-iot-react#1388]
### Describe the bug

If column is NOT sortable, it contains `id=` and `data-columns=` which can be used by automation to find a column.

![image](https://user-images.githubusercontent.com/1056553/87074427-d8aed300-c1ec-11ea-9df2-69fdfffa018f.png)

If the column is sortable then there's no unique way to find the column, since it no longer had `id` or `data-column`

![image](https://user-images.githubusercontent.com/1056553/87074550-085ddb00-c1ed-11ea-80eb-b7468af17d80.png)

I'm unsure if this is a regression, or if it always worked this way.

### To Reproduce

In the description

### Expected behavior

It would be good if there was a unique way to target a column for testing

**Comment by tay1orjones** [comment]
I'm not sure why this is happening, but in any case we should place a `data-testid` on these to meet this need. This is sort of a duplicate of #1001, but will leave open as we can tackle this item independently.

**Comment by bjornalm** [comment]
I just noticed this the other day: the TableHeader.js has two completely different implementations depending on if the header is sortable or not. The non sortable version uses `{...rest}` whereas the sortable implementation only uses explicit properties and `id` and `data-columns` are not among those explicit props.

**Comment by tay1orjones** [comment]
Yeah, that was definitely an oversight in building the sorting functionality. Carbon spreads props in their TableHeader on the `th` 

https://github.com/carbon-design-system/carbon/blob/2692813cc53972ef4a460a51d87b7e45285ca742/packages/react/src/components/DataTable/TableHeader.js#L61
